kotlin.jvm.internal.ArrayLongIterator
<kotlin.jvm.internal.ArrayLongIterator: long[] array>
<kotlin.jvm.internal.ArrayLongIterator: int index>
    public void <init>(long[])
    {
        kotlin.jvm.internal.ArrayLongIterator $r0;
        long[] $r1;

        $r0 := @this: kotlin.jvm.internal.ArrayLongIterator;

        $r1 := @parameter0: long[];

        staticinvoke <kotlin.jvm.internal.Intrinsics: void checkNotNullParameter(java.lang.Object,java.lang.String)>($r1, "array");

        specialinvoke $r0.<kotlin.collections.LongIterator: void <init>()>();

        $r0.<kotlin.jvm.internal.ArrayLongIterator: long[] array> = $r1;

        return;
    }

    public boolean hasNext()
    {
        kotlin.jvm.internal.ArrayLongIterator $r0;
        int $i0, $i1;
        long[] $r1;

        $r0 := @this: kotlin.jvm.internal.ArrayLongIterator;

        $i0 = $r0.<kotlin.jvm.internal.ArrayLongIterator: int index>;

        $r1 = $r0.<kotlin.jvm.internal.ArrayLongIterator: long[] array>;

        $i1 = lengthof $r1;

        if $i0 >= $i1 goto label1;

        return 1;

     label1:
        return 0;
    }

    public long nextLong()
    {
        kotlin.jvm.internal.ArrayLongIterator $r0;
        int $i0, $i1;
        java.lang.String $r1;
        long[] $r2;
        long $l2;
        java.lang.Throwable $r3;
        java.util.NoSuchElementException r4;
        java.lang.ArrayIndexOutOfBoundsException r5;

        $r0 := @this: kotlin.jvm.internal.ArrayLongIterator;

        $r2 = $r0.<kotlin.jvm.internal.ArrayLongIterator: long[] array>;

        $i1 = $r0.<kotlin.jvm.internal.ArrayLongIterator: int index>;

        $i0 = $i1 + 1;

        $r0.<kotlin.jvm.internal.ArrayLongIterator: int index> = $i0;

     label1:
        $l2 = $r2[$i1];

     label2:
        return $l2;

     label3:
        $r3 := @caughtexception;

        $i0 = $r0.<kotlin.jvm.internal.ArrayLongIterator: int index>;

        $i0 = $i0 + -1;

        $r0.<kotlin.jvm.internal.ArrayLongIterator: int index> = $i0;

        r4 = new java.util.NoSuchElementException;

        r5 = (java.lang.ArrayIndexOutOfBoundsException) $r3;

        $r1 = virtualinvoke r5.<java.lang.ArrayIndexOutOfBoundsException: java.lang.String getMessage()>();

        specialinvoke r4.<java.util.NoSuchElementException: void <init>(java.lang.String)>($r1);

        throw r4;

        catch java.lang.ArrayIndexOutOfBoundsException from label1 to label2 with label3;
    }

